world/stat_checks/check_maker.py

```diff
@@ -88,6 +94,68 @@ def announce_to_room(self):
             self.roll_message, options={"roll": True}
         )
 
+    def announce_to_players(self):
+        """
+        Sends a private roll result message to specific players as well as
+        all staff at self.character's location.
+        """
+        self_list = (
+            "me",
+            "self",
+            str(self.character).lower(),
+            str(self.character.key).lower(),
+        )
+
+        # Adjust receiver list; this is who actually will receive it besides staff and the caller.
+        receiver_list = self.receivers.copy()
+        for name in self.receivers:
+            receiver = self.character.search(name.strip(), use_nicks=True)
+
+            # If not found, is the caller, or is a GM, remove from the list of receivers.
+            # (Staff and the caller don't need to get it twice; they always receive it.)
+            if (
+                not receiver
+                or receiver.check_permstring("Builders")
+                or name.lower() in self_list
+                or receiver_list.count(name) > 1
+            ):
+                receiver_list.remove(name)
+
+        # Is this message meant just for me, or am I now the only recipient?
+        self_only = False
+        if len(receiver_list) == 0:
+            receiver_list.append("self-only")
+            self_only = True
+
+        # Now that we know who's getting it, build the private message string.
+        receiver_suffix = "(Shared with: %s)" % (", ").join(receiver_list)
+        private_msg = f"|w[Private Roll]|n {self.roll_message} {receiver_suffix}"
+
+        # Always sent to yourself.
+        self.character.msg(private_msg, options={"roll": True})
+
+        # Send result message to all staff in location.
+        staff_list = [
+            gm
+            for gm in self.character.location.contents
```

Comment:
Similarly, `self.character.location` is not guaranteed to exist. Characters who are offline have no location, and when porting this to actions we'd want to only have this check if they have a location before checking its contents